ci-operator/step-registry/azure/provision/bastionhost/managed-identity/azure-provision-bastionhost-managed-identity-commands.sh

```diff
@@ -0,0 +1,110 @@
+        export REQUESTS_CA_BUNDLE=/tmp/ca.pem
+    fi
+    az cloud register \
+        -n ${cloud_name} \
+        --endpoint-resource-manager "${AZURESTACK_ENDPOINT}" \
+        --suffix-storage-endpoint "${SUFFIX_ENDPOINT}"
+    az cloud set --name ${cloud_name}
+    az cloud update --profile 2019-03-01-hybrid
+else
+    az cloud set --name AzureCloud
+fi
+az login --service-principal -u "${AZURE_AUTH_CLIENT_ID}" -p "${AZURE_AUTH_CLIENT_SECRET}" --tenant "${AZURE_AUTH_TENANT_ID}" --output none
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+bastion_name="${CLUSTER_NAME}-bastion"
+
+if [ -z "${RESOURCE_GROUP}" ]; then
+  rg_file="${SHARED_DIR}/resourcegroup"
+  if [ -f "${rg_file}" ]; then
+    bastion_rg=$(cat "${rg_file}")
+  else
+    echo "Did not find ${rg_file}!"
+    exit 1
+  fi
+else
+  bastion_rg="${RESOURCE_GROUP}"
+fi
+
+command -v az
+az --version
+
+azure_identity_auth_file="${SHARED_DIR}/azure_managed_identity_osServicePrincipal.json"
+if [[ "${AZURE_MANAGED_IDENTITY_TYPE}" == "user-defined" ]]; then
+    # create user-defined managed identity
+    echo "Enable user-defined managed identity on bastion vm"
+    user_identity_name="${CLUSTER_NAME}-identity"
+    run_command "az identity create -g ${bastion_rg} -n ${user_identity_name}"
+
+    # enable user-defined managed identity on bastion vm
+    run_command "az vm identity assign --identities ${user_identity_name} -g ${bastion_rg} -n ${bastion_name}"
+
+    # assign role
+    echo "Assign role 'Contributor' 'User Access Administrator' to the identity"
+    user_identity_id=$(az identity show -g ${bastion_rg} -n ${user_identity_name} --query 'principalId' -otsv)
+    run_command "az role assignment create --role 'Contributor' --assignee ${user_identity_id} --scope /subscriptions/${AZURE_AUTH_SUBSCRIPTION}"
+    run_command "az role assignment create --role 'User Access Administrator' --assignee ${user_identity_id} --scope /subscriptions/${AZURE_AUTH_SUBSCRIPTION}"
+
+    # save azure auth json file
+    user_identity_clientid=$(az vm identity show -g "${bastion_rg}" -n "${bastion_name}" --query "userAssignedIdentities.\"/subscriptions/${AZURE_AUTH_SUBSCRIPTION}/resourceGroups/${bastion_rg}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/${user_identity_name}\".clientId" -otsv)
+    cat > "${azure_identity_auth_file}" << EOF
+{"subscriptionId":"${AZURE_AUTH_SUBSCRIPTION}","clientId":"${user_identity_clientid}","tenantId":"${AZURE_AUTH_TENANT_ID}"}
+EOF
+    
+elif [[ "${AZURE_MANAGED_IDENTITY_TYPE}" == "system" ]]; then
+    # enable system managed identity on bastion
+    # assign "Contributor" and "User Access Administrator role" to identity
+    echo "Enable system managed identity on bastion vm"
+    run_command "az vm identity assign -g ${bastion_rg} -n ${bastion_name} --role Contributor --scope /subscriptions/${AZURE_AUTH_SUBSCRIPTION}"
+    echo "Assign role 'Contributor' 'User Access Administrator' to the identity"
```

Comment:
only 'User Access Administrator' here, right?

Reply:
ah, yes, 'Contributor' role has been assigned in previous command, will update echo message.